app/validate.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING, cast
+
+from checkov.common.bridgecrew.check_type import CheckType
+from checkov.common.bridgecrew.severities import BcSeverities
+from flask import current_app as webhook
+
+from app.consts import UUID_PATTERN
+from app.utils import admission_response
+
+if TYPE_CHECKING:
+    from checkov.common.output.report import Report
+    from flask import Response
+
+    from app.models import CheckovWhorf
+
+
+def process_passed_checks(ckv_whorf: CheckovWhorf, uid: str, obj_kind_name: str) -> Response:
+    """Invoked when no Kubernetes related issues were found"""
+
+    message = []
+    sca_message = []
+
+    for report in ckv_whorf.scan_reports:
+        if report.check_type == CheckType.SCA_IMAGE:
+            sca_message = collect_cves_and_license_violations(report=report)
+
+    message.append("Checkov found 0 total issues in this manifest.")
+    message.extend(sca_message)
+
+    webhook.logger.info(f"Object {obj_kind_name} passed security checks. Allowing the request.")
+    return admission_response(allowed=True, uid=uid, message="\n".join(message))
+
+
+def process_failed_checks(ckv_whorf: CheckovWhorf, uid: str, obj_kind_name: str) -> Response:
+    """Invoked when Kubernetes related issues were found"""
+
+    message = []
+    sca_message = []
+
+    if ckv_whorf.config.hard_fail_on:
+        hard_fails = {}
+        try:
+            for report in ckv_whorf.scan_reports:
+                for check in report.failed_checks:
+                    if check.check_id in ckv_whorf.config.hard_fail_on:
+                        hard_fails[check.check_id] = f"\n  Description: {check.check_name}"
+                        if check.guideline:
+                            hard_fails[check.check_id] += f"\n  Guidance: {check.guideline}"
+                    elif check.bc_check_id in ckv_whorf.config.hard_fail_on:
+                        hard_fails[check.check_id] = f"\n  Description: {check.check_name}"
+                        if check.guideline:
+                            hard_fails[check.check_id] += f"\n  Guidance: {check.guideline}"
+        finally:
+            webhook.logger.error("hard fail error")
+
+        if hard_fails:
+            message.append(f"Checkov found {len(hard_fails)} issues in violation of admission policy.")
+
+            for ckv in hard_fails:
+                message.append(f"{ckv}:{hard_fails[ckv]}")
+
+    issue_count = 0
+    for report in ckv_whorf.scan_reports:
+        if report.check_type == CheckType.SCA_IMAGE:
+            sca_message = collect_cves_and_license_violations(report=report)
```

Comment:
I would extract that logic (it repeats itself in `process_passed_checks`) and put it in a separate context with `collect_cves_and_license_violations` since both are some kind of enrichment on top of the `validate` logic

Reply:
I created now 2 separate functions, which generates the Kubernetes and SCA output, let me know, what you think